app/js/components/tabs/tabcomponents/programmeComponent.jsx

```diff
@@ -1,17 +1,86 @@
+    componentDidMount(props) {
+        this.props.fetchData('/program').then(data => {
+            this.setState({
+                programs: data.results
+            });
+        });
+        this.props.fetchData('/location').then(location => {
+            this.setState({
+                locations: location.results
+            });
+        })
+    }
+    // Get program workflows for the program selected
+    getWorkflow() {
+        const program = document.getElementById("select-program").value;
+        this.props.fetchData(`/program/${program}`).then(data => {
+            this.setState({
+                workflows: data.allWorkflows
+            });
+            this.getStates(data.allWorkflows);
+        });
+    }
+    // Get the states from the workflow.
+    getStates(workflows) {
+        /** Update state based on the data retreived from the workflows
+         * @TODO: State data does not exist on the local, but exists on refapp,
+         * try to populate states on the local
+        */
+        
+    }
+    render() {
+        let programs = this.state.programs.map((program) => {
+            return (
+                <option key={program.uuid} value={program.uuid}>
+                    {program.name}
+                </option>
+            );
+        });
+        let workflows = this.state.workflows.map((workflow) => {
+            return (
+                <option key={workflow.uuid} value={workflow.uuid}>
+                    {workflow.concept.display}
+                </option>
+            );  
+        })
+        // States will be loaded from this.state.states when populated from backend
+        let states = "<option> </option>";
+
+        let locations = this.state.locations.map((location) => {
+            return (
+                <option key={location.uuid} value={location.uuid}>
+                    {location.display}
+                </option>        
+            )
+        })
+
+        // Methods will be loadd from this.state.methods populated from the backend.
+        let methods = "<option> </option>";
     return (
         <div className="programme-component">
             <h3>Search By Program Enrollement and Status</h3>
             <form className="form-horizontal">
                 <div className="form-group">
                     <label htmlFor="gender" className="col-sm-2 control-label">Program:</label>
                     <div className="col-sm-6">
-                        <select className="form-control" id="" name="">
+                        <select onChange={() => {this.getWorkflow()}} className="form-control" id="select-program" name="">
                             <option value="all">All</option>
-                            <option value="male">Program 1</option>
-                            <option value="female">Program 2</option>
+                            {/* Display the registered programs */}
```

Comment:
They're not really necessary, just a way to keep track of where the backend data were added, I will go ahead and remove them.